### PR TITLE
feature(component): carousel 22

### PR DESCRIPTION
**Pull request by codeuri**
- slider
- slider bar
- timing
- tests
- stories
- docs

### CODE REVIEW

**Comment by gepifanio** [comment]
Comment: Deploy Started

Building with commit a995eaf05a17bb1e62b936633574744948f17c9c

https://app.netlify.com/sites/pattern-lib-unimelb/deploys/5cd5a20e579d730008dc6a42

File: components/carousel/Carousel.vue [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { slider, slideritem } from 'vue-concise-slider';
+import SvgIcon from '../icons/SvgIcon.vue';
+
+const autoplay = 5000;
```
Comment (gepifanio) [inline components/carousel/Carousel.vue]: The carousel should automatically cycle through its stories every 20 seconds

File: components/carousel/stories/carousel-doc.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    buttonHref: 'google.com',
+  },
+]'
+```
+
+Usage:
+
+```html
+<carousel
+ s  stories="[
```
Comment (gepifanio) [inline components/carousel/stories/carousel-doc.md]: typo

**Comment by gepifanio** [comment]
Hey @anormand,

Could you check the Carousel in this link: https://deploy-preview-1066--pattern-lib-unimelb.netlify.com/?selectedKind=Carousel&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=REACT_STORYBOOK%2Freadme%2Fpanel

And make sure is following to accessibility requirements? 

Thank you!

File: components/carousel/Carousel.vue [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    <div
+      class="carousel--panel"
+    >
+      <div
+        :style="{ width: progressBarWidth + '%'}"
+        class="carousel--panel__progress-bar"/>
+      <div class="carousel--panel__container">
+        <div class="carousel--panel__left">
+          <div>
+            <div class="carousel--panel--title">
```
Comment (anormand) [inline components/carousel/Carousel.vue]: Story title should be ```<h2>```

File: components/carousel/Carousel.vue [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+          </div>
+          <ButtonIcon
+            :href="selectedItem.buttonHref"
+            width="wide"
+            class="carousel--panel--cta"
+          >{{ selectedItem.buttonText }}</ButtonIcon>
+        </div>
+        <div class="carousel--panel__right">
+          <div class="carousel--panel--controls">
+            <div
```
Comment (gepifanio) [inline components/carousel/Carousel.vue]: This div should read to the screen reader = "Previous story + Story title"

File: components/carousel/Carousel.vue [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+              @click="startSliding"
+              @keydown.13="startSliding"
+            >
+              <SvgIcon
+                name="play"
+                width="16"
+                height="16"
+              />
+            </div>
+            <div
```
Comment (gepifanio) [inline components/carousel/Carousel.vue]: This div should read to the screen reader = "Next story + Story title"

File: components/carousel/Carousel.vue [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+              @click="prevStory"
+              @keydown.13="prevStory"
+            >
+              <SvgIcon
+                name="chevron-left"
+                width="16"
+                height="16"
+              />
+            </div>
+            <div
```
Comment (gepifanio) [inline components/carousel/Carousel.vue]: This div should read to the screen reader = "Pause"

File: components/carousel/Carousel.vue [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+              @click="stopSliding"
+              @keydown.13="stopSliding"
+            >
+              <SvgIcon
+                name="pause"
+                width="16"
+                height="16"
+              />
+            </div>
+            <div
```
Comment (gepifanio) [inline components/carousel/Carousel.vue]: This div should read to the screen reader = "Play"

File: components/carousel/Carousel.vue [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+              @keydown.13="nextStory"
+            >
+              <SvgIcon
+                name="chevron-right"
+                width="16"
+                height="16"
+              />
+            </div>
+          </div>
+          <div class="carousel--panel--stories-menu">
```
Comment (gepifanio) [inline components/carousel/Carousel.vue]: The markup for this menu/tabs bar should come first, so the people with disabilities get first in the menu, however, the design still the same.

File: components/carousel/Carousel.vue [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+            <li
+              v-for="(story, index) in storiesData"
+              :key="index"
+              tabindex="0"
+              class="carousel--panel--stories-menu--item"
+              @click="moveToStory(index)"
```
Comment (gepifanio) [inline components/carousel/Carousel.vue]: If someone clicks in the story, the keyboard focus should go the story title/link.

File: components/carousel/index.css [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+    &__right {
+      flex-basis: 30%;
+      display: flex;
+      flex-direction: column;
+      justify-content: space-around;
+      margin-left: 1rem;
+    }
+
+    &--title {
+      color: #094183;
```
Comment (TimMansell) [inline components/carousel/index.css]: --col-navy-dark

File: components/carousel/index.css [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+  &--footer {
+    height: 115px;
+    width: 100%;
+    background: rgb(var(--col-background-primary-navy));
+  }
+
+  &--panel {
+    width: 75%;
+    margin: auto;
+    background: #f2f5f9;
```
Comment (TimMansell) [inline components/carousel/index.css]: --col-navy-light

File: components/carousel/index.css [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+      flex-direction: column;
+      justify-content: space-around;
+      margin-left: 1rem;
+    }
+
+    &--title {
+      color: #094183;
+      font-weight: var(--fw-bold);
+      font-size: 1.35rem;
+      line-height: 27px;
```
Comment (TimMansell) [inline components/carousel/index.css]: use unitless value: `line-height: 1;`

File: components/carousel/index.css [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+      margin-bottom: 1rem;
+    }
+
+    &--description {
+      color: #000;
+      font-size: 1rem;
+      line-height: 24px;
```
Comment (TimMansell) [inline components/carousel/index.css]: use unitless value: `line-height: 1.35;`

File: components/carousel/index.css [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    &--cta {
+      max-width: 20%;
+      background: #fff;
```
Comment (TimMansell) [inline components/carousel/index.css]: --col-background-primary-white

File: components/carousel/index.css [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+      border-color: rgb(var(--col-background-secondary-grey));
+
+      span {
+        color: #000;
```
Comment (TimMansell) [inline components/carousel/index.css]: --col-black-dark

File: components/carousel/index.css [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+
+    &__progress-bar {
+      height: 4px;
+      background: rgb(var(--col-background-primary-navy));
+      transition: width 200ms linear;
+    }
+
+    &--controls {
+      display: flex;
+      color: #000;
```
Comment (TimMansell) [inline components/carousel/index.css]: --col-black-dark

File: components/carousel/index.css [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    &--stop-pause {
+      margin: 0 1rem;
+    }
+
+    &--stories-menu {
+      &--item {
+        list-style-type: none;
+        border-bottom: 1px solid #99a2aa;
```
Comment (TimMansell) [inline components/carousel/index.css]: --col-black-mid-2

File: components/carousel/index.css [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+      &--text {
+        padding: .3rem .4rem;
+        color: rgb(var(--col-background-primary-navy));
+        font-weight: var(--fw-bold);
+        font-size: 1.2rem;
+      }
+
+      &__active {
+        background: rgb(var(--col-background-primary-navy));
+        color: #fff;
```
Comment (TimMansell) [inline components/carousel/index.css]: --col-background-primary-white

File: components/carousel/index.css [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+
+.carousel {
+  display: flex;
+  justify-content: center;
+  flex-direction: column;
+  align-items: center;
+  width: 100%;
+  position: relative;
+
+  &--slider {
```
Comment (gepifanio) [inline components/carousel/index.css]: Should be an element instead of modifier
`__slider`

File: components/carousel/index.css [hunk 20]
```diff
@@ -0,0 +1,187 @@
...
+      height: 548px;
+    }
+
+    &__image {
+      height: 100%;
+      width: 100%;
+    }
+  }
+
+  &--footer {
```
Comment (gepifanio) [inline components/carousel/index.css]: Should be an element instead of modifier
__footer

File: components/carousel/index.css [hunk 21]
```diff
@@ -0,0 +1,187 @@
...
+    display: none;
+    @media (--bp-desktop) {
+      display: block;
+      height: 115px;
+      width: 100%;
+      background: rgb(var(--col-background-primary-navy));
+    }
+  }
+
+  &--panel {
```
Comment (gepifanio) [inline components/carousel/index.css]: Should be an element instead of modifier
`__panel`

File: components/carousel/index.css [hunk 22]
```diff
@@ -0,0 +1,187 @@
...
+    margin: auto;
+  }
+
+  @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
+    display: block;
+    left: 10vw;
+  }
+
+
+    &__left {
```
Comment (gepifanio) [inline components/carousel/index.css]: should be a modifier `--left`

File: components/carousel/index.css [hunk 23]
```diff
@@ -0,0 +1,187 @@
...
+      display: flex;
+      flex: 1;
+      flex-direction: column;
+      justify-content: space-between;
+    }
+
+    &__right {
```
Comment (gepifanio) [inline components/carousel/index.css]: should be a modifier `--left`

File: components/carousel/index.css [hunk 24]
```diff
@@ -0,0 +1,187 @@
...
+      display: flex;
+      justify-content: space-around;
+      @media (--bp-desktop) {
+        flex-basis: 30%;
+        margin-left: 1rem;
+        flex-direction: column-reverse;
+      }
+    }
+
+    &--story {
```
Comment (gepifanio) [inline components/carousel/index.css]: should be an element `__story`

File: components/carousel/index.css [hunk 25]
```diff
@@ -0,0 +1,187 @@
...
+      overflow: scroll;
+    }
+
+    &--title {
```
Comment (gepifanio) [inline components/carousel/index.css]: should be an element `__title`

File: components/carousel/index.css [hunk 26]
```diff
@@ -0,0 +1,187 @@
...
+      font-size: 1.35rem;
+      a {
+        color: rgb(var(--col-navy-dark));
+        font-weight: var(--fw-bold);
+        line-height: 1;
+        display: block;
+      }
+    }
+
+    &--description {
```
Comment (gepifanio) [inline components/carousel/index.css]: should be an element `__description`

File: components/carousel/index.css [hunk 27]
```diff
@@ -0,0 +1,187 @@
...
+        list-style-type: none;
+        border-bottom: 1px solid rgb(var(--col-black-mid-2));
+        padding: .42rem 0;
+        margin: 0;
+        cursor: pointer;
+        display: block;
+        text-decoration: none;
+      }
+
+      &--text {
```
Comment (gepifanio) [inline components/carousel/index.css]: should be an element `__text`

File: components/carousel/stories/carousel-doc.md [hunk 28]
```diff
@@ -0,0 +1,54 @@
+## Carousel
```
Comment (gepifanio) [inline components/carousel/stories/carousel-doc.md]: the story does not show properly how to use the component and do not contain the information about the timing for the stories and what are the options.

File: components/carousel/Carousel.vue [hunk 29]
```diff
@@ -0,0 +1,297 @@
...
+              <span
+                :class="{ 'carousel--panel--stories-menu__active': story.isActive }"
+                class="carousel--panel--stories-menu--text">
+                {{ story.title }}
+              </span>
+            </a>
+          </div>
+          <div class="carousel--panel--controls">
+            <div
+              :area-label="'Previous: ' + prevTitle"
```
Comment (TimMansell) [inline components/carousel/Carousel.vue]: typo: aria-label

File: components/carousel/Carousel.vue [hunk 30]
```diff
@@ -0,0 +1,297 @@
...
+            >
+              <SvgIcon
+                name="play"
+                width="16"
+                height="16"
+                aria-hidden="true"
+              />
+            </div>
+            <div
+              :area-label="'Next: ' + nextTitle"
```
Comment (TimMansell) [inline components/carousel/Carousel.vue]: typo: aria-label

File: components/carousel/Carousel.vue [hunk 31]
```diff
@@ -0,0 +1,297 @@
...
+      class="carousel--panel"
+    >
+      <div
+        :style="{ width: progressBarWidth + '%'}"
+        class="carousel--panel__progress-bar"/>
+      <div class="carousel--panel__container">
+        <div
+          class="carousel--panel__right"
+          role="navigation"
+          area-label="Stories"
```
Comment (TimMansell) [inline components/carousel/Carousel.vue]: typo: aria-label

File: components/carousel/Carousel.vue [hunk 32]
```diff
@@ -0,0 +1,297 @@
...
+                aria-hidden="true"
+              />
+            </div>
+          </div>
+        </div>
+        <div
+          :aria-live="!paused ? 'off' : 'polite'"
+          class="carousel--panel__left"
+          role="navigation"
+          area-label="Stories">
```
Comment (TimMansell) [inline components/carousel/Carousel.vue]: typo: aria-label

**Comment by codeuri** [comment]
Hi @TimMansell and @gepifanio 
All the above is done, but I created new PR -> https://github.com/unimelb/pattern-lib/pull/1183
I created new PR because of too many conflicts.

Cheers

File: components/carousel/Carousel.vue [hunk 33]
```diff
@@ -0,0 +1,226 @@
...
+      <div
+        :style="{ width: progressBarWidth + '%'}"
+        class="carousel--panel__progress-bar"/>
+      <div class="carousel--panel__container">
+        <div class="carousel--panel__left">
+          <div>
+            <div class="carousel--panel--title">
+              {{ selectedItem.title }}
+            </div>
+            <div class="carousel--panel--description">
```
[the remark this answers is not in the document]
Comment (TimMansell) [reply components/carousel/Carousel.vue]: Looks good.

File: components/carousel/Carousel.vue [hunk 34]
```diff
@@ -0,0 +1,226 @@
...
+              {{ selectedItem.description }}
+            </div>
+          </div>
+          <ButtonIcon
```
[the remark this answers is not in the document]
Comment (TimMansell) [reply components/carousel/Carousel.vue]: Looks good.

File: components/carousel/Carousel.vue [hunk 35]
```diff
@@ -0,0 +1,226 @@
...
+          <div class="carousel--panel--stories-menu">
+            <li
+              v-for="(story, index) in storiesData"
+              :key="index"
+              tabindex="0"
+              class="carousel--panel--stories-menu--item"
+              @click="moveToStory(index)"
+              @keydown.13="moveToStory(index)"
+            >
+              <div
```
[the remark this answers is not in the document]
Comment (TimMansell) [reply components/carousel/Carousel.vue]: Hi @anormand can you check this in the new PR #1183

File: components/carousel/Carousel.vue (hunk 9 above)
[the remark this answers is not in the document]
Comment (TimMansell) [reply components/carousel/Carousel.vue]: Hi @anormand can you check this in the new PR #1183

File: components/carousel/Carousel.vue [hunk 36]
```diff
@@ -0,0 +1,226 @@
...
+            </div>
+          </div>
+          <div class="carousel--panel--stories-menu">
+            <li
+              v-for="(story, index) in storiesData"
+              :key="index"
+              tabindex="0"
+              class="carousel--panel--stories-menu--item"
+              @click="moveToStory(index)"
+              @keydown.13="moveToStory(index)"
```
[the remark this answers is not in the document]
Comment (TimMansell) [reply components/carousel/Carousel.vue]: Hi @anormand can you check this in the new PR #1183

File: components/carousel/Carousel.vue [hunk 37]
```diff
@@ -0,0 +1,226 @@
...
+    </div>
+    <div class="carousel--footer" />
+    <div
+      class="carousel--panel"
+    >
+      <div
+        :style="{ width: progressBarWidth + '%'}"
+        class="carousel--panel__progress-bar"/>
+      <div class="carousel--panel__container">
+        <div class="carousel--panel__left">
```
[the remark this answers is not in the document]
Comment (TimMansell) [reply components/carousel/Carousel.vue]: Hi @anormand can you check this in the new PR #1183

File: components/carousel/Carousel.vue [hunk 38]
```diff
@@ -0,0 +1,226 @@
...
+                width="16"
+                height="16"
+              />
+            </div>
+          </div>
+          <div class="carousel--panel--stories-menu">
+            <li
+              v-for="(story, index) in storiesData"
+              :key="index"
+              tabindex="0"
```
[the remark this answers is not in the document]
Comment (TimMansell) [reply components/carousel/Carousel.vue]: Hi @anormand can you check this in the new PR #1183

File: components/carousel/Carousel.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (TimMansell) [reply components/carousel/Carousel.vue]: Hi @anormand can you check this in the new PR #1183

File: components/carousel/Carousel.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (TimMansell) [reply components/carousel/Carousel.vue]: Hi @anormand can you check this in the new PR #1183

File: components/carousel/Carousel.vue [hunk 39]
```diff
@@ -0,0 +1,226 @@
...
+              {{ selectedItem.description }}
+            </div>
+          </div>
+          <ButtonIcon
+            :href="selectedItem.buttonHref"
+            width="wide"
+            class="carousel--panel--cta"
+          >{{ selectedItem.buttonText }}</ButtonIcon>
+        </div>
+        <div class="carousel--panel__right">
```
[the remark this answers is not in the document]
Comment (TimMansell) [reply components/carousel/Carousel.vue]: Hi @anormand can you check this in the new PR #1183

**Pull request closed without merging**